src/PromoService.kt

```
@@ -0,0 +1,1342 @@
+        }
+        createPromoDto.promo?.promoTargetProductType?.let {
+            userPromo.promoTargetProductType = it
+        }
+
+        try {
+            userPromoRepository.save(userPromo).also {
+                processUserPromo(userPromo)
+            }
+            logger.info { "promo ${createPromoDto.code} saved for user: ${createPromoDto.userId}" }
+            return userPromo
+        } catch (e: DataIntegrityViolationException) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)